### ISSUE TITLE
Button props in hook should extends default props

### ISSUE THREAD

**Issue by lkostrowski** [jonatanklosko/material-ui-confirm#16]
Given 

```
<ConfirmProvider
			defaultOptions={{
				confirmationButtonProps: {
					variant: 'contained',
					disableElevation: true,
				},
			}}
		>
```

And

```
confirm({
			title: 'Title',
			confirmationButtonProps: {
				disabled: true,
			},
		})
```

**Expected:**

Confirm button will have props:

```
{
  variant: 'contained',
  disableElevation: true,
  disabled: true,
}

```

**Actual**

Props get overwritten and compute to 

```
{
  disabled: true,
}

```

**Comment by lkostrowski** [comment]
https://github.com/jonatanklosko/material-ui-confirm/blob/master/src/ConfirmProvider.js#L22

I see that in theory it should work. I will look for a reproduction

**Comment by jonatanklosko** [comment]
Actually options are merged in a shallow way, so setting global `confirmationButtonProps: X` then calling with `confirmationButtonProps: Y` results in object `Y` being the actual value. I see how deep merging would make sense here, although then it would be hard to override the default options (we'd need to explicitly override each unwanted default option with undefined or so). Not sure what makes most sense, would you say deep merging is preferable in this context?

**Comment by lkostrowski** [comment]
I think shallow merge is fine, but it should be done to each options, not the general options. So basically it's enough to go 1 level deeper, but not recursive.

So `const confirmationButtonProps = {..._defaultOptions.confirmationButtonProps, ...options.confirmationButtonProps, ...actualOptions.confirmationButtonProps}` etc

Also the best api would be if you allow both object and mapping func so:

`confirmationButtonProps: ButtonProps | (defaultProps: ButtonProps): ButtonProps`

Because in some cases you want to check what was the default in the context

**Comment by jonatanklosko** [comment]
Yeah, 1 level definitely sounds enough.

> Because in some cases you want to check what was the default in the context

Hmm, I'd say the default context is your global setting. It could only differ if you'd have different providers in multiple places and reused the same component somewhere inside their tree (which seems unnecessarily complicated to me). So I don't really see how such function could be useful.

**Comment by lkostrowski** [comment]
Yea, its more like "high level api design" to provide more control over these settings - but you are right that this case is not something common to use - it can be added any time later if needed.

So agree - 1 level deeper than now

**Comment by jonatanklosko** [comment]
I've just pushed 1acedcb33c00a6fccd32b61c8d0c944c68297bbb, please have a look at the test and let me know if it looks as expected =)

**Comment by lkostrowski** [comment]
Can you publish it on npm?

**Comment by jonatanklosko** [comment]
Sure, published as `2.1.0`.

**Comment by lkostrowski** [comment]
Works like a  charm. Thanks!
